pkg/validation/validation.go

```diff
@@ -316,3 +348,85 @@ func supportsVersionUpgrade(uc *upgradev1alpha1.UpgradeConfig) bool {
 func empty(s string) bool {
 	return strings.TrimSpace(s) == ""
 }
+
+// fetchImageVersion function returns the image version from the image digest
+func fetchImageVersion(image string) (string, error) {
+	ref, _ := imagereference.Parse(image)
+	manifesturl := url.URL{
+		Scheme: "https",
+		Host:   ref.Registry,
+		Path:   "v2" + "/" + ref.Namespace + "/" + ref.Name + "/" + "manifests" + "/" + ref.ID,
+	}
+
+	body, err := runHTTP(manifesturl.String())
+	if len(body) == 0 {
+		return "", fmt.Errorf("Failed to fetch image manifest: %s  may not be a valid image", image)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	manifest := &dockerv1client.DockerImageManifest{}
+	if err := parse(body, &manifest); err != nil {
+		return "", err
+	}
+	bloburl := url.URL{
+		Scheme: "https",
+		Host:   ref.Registry,
+		Path:   "v2" + "/" + ref.Namespace + "/" + ref.Name + "/" + "blobs" + "/" + manifest.Config.Digest,
+	}
+
+	resbody, err := runHTTP(bloburl.String())
+	if len(resbody) == 0 {
+		return "", fmt.Errorf("Failed to fetch image imageconfig for %s", image)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	imageConfig := &dockerv1client.DockerImageConfig{}
+	if err := parse(resbody, &imageConfig); err != nil {
+		return "", err
+	}
+	return imageConfig.Config.Labels["io.openshift.release"], nil
+}
+
+func runHTTP(url string) ([]byte, error) {
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req.Header.Set("Accept", "application/vnd.docker.distribution.manifest.v2+json")
+	client := http.Client{
+		Timeout: time.Second * 20,
+	}
+
+	res, getErr := client.Do(req)
+	if getErr != nil {
+		return nil, getErr
+	}
+
+	if res.Status != "200 OK" {
```

Comment:
Better to compare `res.StatusCode` to `http.StatusOK`.

Reply:
@mrbarge Thank you for your review. Applied the suggested change.